### ISSUE TITLE
websocket_refactor branch error upon `make run`

### ISSUE THREAD

**Issue by todpole3** [TellinaTool/tellina_task_interface#3]
The `npm` error below was encountered:
```
make run
sudo -H pip3 install -r ~/tellina_task_interface/requirements.txt
Requirement already satisfied (use --upgrade to upgrade): Django==1.10.4 in /usr/local/lib/python3.5/dist-packages (from -r /home/vagrant/tellina_task_interface/requirements.txt (line 1))
Requirement already satisfied (use --upgrade to upgrade): docker-py==1.10.6 in /usr/local/lib/python3.5/dist-packages (from -r /home/vagrant/tellina_task_interface/requirements.txt (line 2))
Requirement already satisfied (use --upgrade to upgrade): django-extensions==1.7.5 in /usr/local/lib/python3.5/dist-packages (from -r /home/vagrant/tellina_task_interface/requirements.txt (line 3))
Requirement already satisfied (use --upgrade to upgrade): websocket-client>=0.32.0 in /usr/local/lib/python3.5/dist-packages (from docker-py==1.10.6->-r /home/vagrant/tellina_task_interface/requirements.txt (line 2))
Requirement already satisfied (use --upgrade to upgrade): docker-pycreds>=0.2.1 in /usr/local/lib/python3.5/dist-packages (from docker-py==1.10.6->-r /home/vagrant/tellina_task_interface/requirements.txt (line 2))
Requirement already satisfied (use --upgrade to upgrade): requests!=2.11.0,>=2.5.2 in /usr/lib/python3/dist-packages (from docker-py==1.10.6->-r /home/vagrant/tellina_task_interface/requirements.txt (line 2))
Requirement already satisfied (use --upgrade to upgrade): six>=1.4.0 in /usr/lib/python3/dist-packages (from docker-py==1.10.6->-r /home/vagrant/tellina_task_interface/requirements.txt (line 2))
You are using pip version 8.1.1, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
sudo docker build -t proxy proxy_image
Sending build context to Docker daemon  5.12 kB
Step 1 : FROM node:6-onbuild
# Executing 5 build triggers...
Step 1 : ARG NODE_ENV
 ---> Using cache
Step 1 : ENV NODE_ENV $NODE_ENV
 ---> Using cache
Step 1 : COPY package.json /usr/src/app/
 ---> Using cache
Step 1 : RUN npm install
 ---> Running in 001414e203bc
npm info it worked if it ends with ok
npm info using npm@3.10.9
npm info using node@v6.9.2
npm info attempt registry request try #1 at 1:43:40 PM
npm http request GET https://registry.npmjs.org/ws
npm info attempt registry request try #1 at 1:43:40 PM
npm http request GET https://registry.npmjs.org/pty.js
npm http 200 https://registry.npmjs.org/pty.js
npm http 200 https://registry.npmjs.org/ws
npm info retry fetch attempt 1 at 1:43:40 PM
npm info attempt registry request try #1 at 1:43:40 PM
npm http fetch GET https://registry.npmjs.org/pty.js/-/pty.js-0.3.1.tgz
npm info retry fetch attempt 1 at 1:43:40 PM
npm info attempt registry request try #1 at 1:43:40 PM
npm http fetch GET https://registry.npmjs.org/ws/-/ws-1.1.1.tgz
npm http fetch 200 https://registry.npmjs.org/pty.js/-/pty.js-0.3.1.tgz
npm http fetch 200 https://registry.npmjs.org/ws/-/ws-1.1.1.tgz
npm info attempt registry request try #1 at 1:43:41 PM
npm http request GET https://registry.npmjs.org/extend
npm info attempt registry request try #1 at 1:43:41 PM
npm http request GET https://registry.npmjs.org/nan
npm http 200 https://registry.npmjs.org/extend
npm info retry fetch attempt 1 at 1:43:41 PM
npm info attempt registry request try #1 at 1:43:41 PM
npm http fetch GET https://registry.npmjs.org/extend/-/extend-1.2.1.tgz
npm http 200 https://registry.npmjs.org/nan
npm info retry fetch attempt 1 at 1:43:41 PM
npm info attempt registry request try #1 at 1:43:41 PM
npm http fetch GET https://registry.npmjs.org/nan/-/nan-2.3.5.tgz
npm http fetch 200 https://registry.npmjs.org/extend/-/extend-1.2.1.tgz
npm http fetch 200 https://registry.npmjs.org/nan/-/nan-2.3.5.tgz
npm info attempt registry request try #1 at 1:43:41 PM
npm http request GET https://registry.npmjs.org/options
npm info attempt registry request try #1 at 1:43:41 PM
npm http request GET https://registry.npmjs.org/ultron
npm http 200 https://registry.npmjs.org/options
npm http 200 https://registry.npmjs.org/ultron
npm info retry fetch attempt 1 at 1:43:41 PM
npm info attempt registry request try #1 at 1:43:41 PM
npm http fetch GET https://registry.npmjs.org/options/-/options-0.0.6.tgz
npm info retry fetch attempt 1 at 1:43:41 PM
npm info attempt registry request try #1 at 1:43:41 PM
npm http fetch GET https://registry.npmjs.org/ultron/-/ultron-1.0.2.tgz
npm http fetch 200 https://registry.npmjs.org/options/-/options-0.0.6.tgz
npm http fetch 200 https://registry.npmjs.org/ultron/-/ultron-1.0.2.tgz
npm info lifecycle undefined~preinstall: undefined
npm info lifecycle extend@1.2.1~preinstall: extend@1.2.1
npm info lifecycle nan@2.3.5~preinstall: nan@2.3.5
npm info lifecycle options@0.0.6~preinstall: options@0.0.6
npm info lifecycle ultron@1.0.2~preinstall: ultron@1.0.2
npm info lifecycle pty.js@0.3.1~preinstall: pty.js@0.3.1
npm info lifecycle ws@1.1.1~preinstall: ws@1.1.1
npm info linkStuff extend@1.2.1
npm info linkStuff nan@2.3.5
npm info linkStuff options@0.0.6
npm info linkStuff ultron@1.0.2
npm info linkStuff pty.js@0.3.1
npm info linkStuff ws@1.1.1
npm info lifecycle extend@1.2.1~install: extend@1.2.1
npm info lifecycle nan@2.3.5~install: nan@2.3.5
npm info lifecycle options@0.0.6~install: options@0.0.6
npm info lifecycle ultron@1.0.2~install: ultron@1.0.2
npm info lifecycle pty.js@0.3.1~install: pty.js@0.3.1

> pty.js@0.3.1 install /usr/src/app/node_modules/pty.js
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp info using node-gyp@3.4.0
gyp info using node@6.9.2 | linux | x64
gyp http GET https://nodejs.org/download/release/v6.9.2/node-v6.9.2-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v6.9.2/node-v6.9.2-headers.tar.gz
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: read ECONNRESET
gyp ERR! stack     at exports._errnoException (util.js:1022:11)
gyp ERR! stack     at TLSWrap.onread (net.js:569:26)
gyp ERR! System Linux 4.4.0-51-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/src/app/node_modules/pty.js
gyp ERR! node -v v6.9.2
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
npm info lifecycle pty.js@0.3.1~install: Failed to exec install script
/usr/src/app
`-- ws@1.1.1
  +-- options@0.0.6
  `-- ultron@1.0.2

npm WARN app No description
npm WARN app No repository field.
npm WARN app No license field.
npm ERR! Linux 4.4.0-51-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v6.9.2
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE

npm ERR! pty.js@0.3.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the pty.js@0.3.1 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the pty.js package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs pty.js
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls pty.js
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /usr/src/app/npm-debug.log
The command '/bin/sh -c npm install' returned a non-zero code: 1
Makefile:19: recipe for target 'build_images' failed
make: *** [build_images] Error 1
```

**Comment by kvu787** [comment]
TLDR:

* Seems like a download issue, try `make run` again.
* If it still fails, try `vagrant destroy -f && vagrant up` and `make run`

The key part of the error is:

```
sudo docker build -t proxy proxy_image
...
gyp http GET https://nodejs.org/download/release/v6.9.2/node-v6.9.2-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v6.9.2/node-v6.9.2-headers.tar.gz
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: read ECONNRESET
gyp ERR! stack     at exports._errnoException (util.js:1022:11)
gyp ERR! stack     at TLSWrap.onread (net.js:569:26)
```

When Docker builds `proxy_image/Dockerfile`, it runs `npm install` which pings `https://nodejs.org` to install Node.js packages. Looks like one of those download calls failed (`ECONNRESET`) and a dependency didn't get installed, so the Docker build failed.

**Comment by kvu787** [comment]
@todpole3 Are you still running into this issue?

**Comment by todpole3** [comment]
@kvu787 It's gone. Thanks!

**Comment by todpole3** [comment]
This happened again today. 

Restart vagrant is not effective.

**Comment by todpole3** [comment]
The error is suppressed by using a different node image: replace `FROM node:6-onbuild` with `FROM nodesource/trusty:5.1` in `proxy_image/Dockerfile` and `backend_container_image/Dockerfile`.
